### PR TITLE
Unittests

### PR DESCRIPTION
**Pull request by matkrak**
Sorry for the mess with the repository. I was trying to work with git console and bad things happened.

Please check unittests.py file where I put some tests for Curve class.
I am writing tests for Profiles objects right now.
-  There is probably better way to compare domains instead of listing them.
- If I rebased this branch you probably wouldn't see this trash commits, am I right?

### CODE REVIEW

**Comment by grzanka** [comment]
I see commit with "vol1 elo" message and I am curious what was the reason to name it like that.

**Comment by grzanka** [comment]
Be more elaborate with tests - cover more corner cases. Add profile methods tests. You can also split tests among more files. Check setup and teardown method. Test is exception machinery works properly.

**Comment by matkrak** [comment]
About this commits - I thought it was something I was doing locally. I made a lot of different commits just to see the results but wasn't aware some of them can be seen by the others - sorry! 
Working on the tests right now.

File: beprof/unittests.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        self.assertTrue(np.isnan(pro.x_at_y(5)))
+
+    def test_mirror(self):
+        # mirroring twice should be like non mirroring at all
+        orgx = copy.deepcopy(lpr.x)
+        orgy = copy.deepcopy(lpr.y)
+        lpr.mirror()
+        lpr.mirror()
+        self.assertEqual(list(lpr.x), list(orgx))
+        self.assertEqual(list(lpr.y), list(orgy))
```
Comment (grzanka) [inline beprof/unittests.py]: Investigate following numpy functions: array_equal, array_equiv, allclose

File: beprof/unittests.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            )
+lpr = LateralProfile([[0, 0], [1, 1], [2, 2], [3, 2], [4, 3], [5, 4], [6, 4], [7, 3], [8, 2], [9, 1], [10, 0]],
+            lprof1 = 'first_lp',
+            lprof2 = 'second_lp'
+            )
+
+
+class TestCurveMethods(unittest.TestCase):
+
+    def test_metadata_1(self):
```
Comment (grzanka) [inline beprof/unittests.py:23]: I think this 3 checks in test_metadata_1 should go to 3 different functions. If the test fail it would be easier to find out why.

File: beprof/unittests.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        # new domain's range not inside current
+        self.assertEqual(list(cur.change_domain([-1, 4]).x), list(cur.x))
+        self.assertEqual(list(cur.change_domain([0, 11]).x), list(cur.x))
+
+    def test_rebinned(self):
+        # different combinations
+        self.assertEqual(list(cur.rebinned(step=1, fixp=0.5).x),
+                         [0.5, 1.5, 2.5, 3.5, 4.5, 5.5, 6.5, 7.5, 8.5, 9.5]
+                         )
+class TestProfileMethods(unittest.TestCase):
```
Comment (grzanka) [inline beprof/unittests.py]: It is very similar to TestCurve methods and seems to violate DRY

**Comment by matkrak** [comment]
Merging subtracting curves PR would provide more functions for testing purpose.
I will also add more malicious tests then.

**Comment by grzanka** [comment]
Leave a message when its done

**Comment by grzanka** [comment]
Closing, as promised. In case of resuming activity, open new PR.
